packages/flutter/lib/src/material/tooltip.dart

```diff
@@ -191,12 +191,6 @@ class Tooltip extends StatefulWidget {
   }) : assert(
          (message == null) != (richMessage == null),
          'Either `message` or `richMessage` must be specified',
-       ),
-       assert(
-         richMessage == null || textStyle == null,
```

Comment:
Nit: Would it be useful to add a test that checks that passing passing both richMessage and textStyle no longer causes an assertion error?

Reply:
This is indirectly covered by the new `Tooltip should pass its default text style down to widget spans` test.

```dart
...
  home: Tooltip(
    key: tooltipKey,
    richMessage: const WidgetSpan(child: Text(tooltipText)),
    textStyle: const TextStyle(fontSize: 4.0, debugLabel: 'size'),
    child: Container(width: 100.0, height: 100.0, color: Colors.green[500]),
  ),
...
```